src/commands/help.ts

```diff
@@ -1,27 +1,91 @@
-const fs = require('fs');
-const path = require('path');
-import { CommandInteraction } from 'discord.js';
+import { CommandInteraction, ActionRowBuilder, ButtonBuilder, ButtonStyle, ButtonInteraction } from 'discord.js';
 import { Command, colors } from '../types';
+import * as fs from 'fs';
+import * as path from 'path';
+
+const ITEMS_PER_PAGE = 8;
 
 export const helpCommand = async (interaction: CommandInteraction) => {
     if (interaction.commandName === "help") {
         const commandsFilePath = path.join(__dirname, '../../commands.json');
         const commands: Command[] = JSON.parse(fs.readFileSync(commandsFilePath, 'utf8'));
 
-        const fields = commands.map((cmd: Command) => ({
+        const totalPages = Math.ceil(commands.length / ITEMS_PER_PAGE);
+
+        const fields = commands.slice(0, ITEMS_PER_PAGE).map((cmd: Command) => ({
             name: `/${cmd.name}`,
             value: cmd.description
         }));
 
+        const row = new ActionRowBuilder<ButtonBuilder>()
+            .addComponents(
+                new ButtonBuilder()
+                    .setCustomId('previous_page')
+                    .setLabel('Previous')
+                    .setStyle(ButtonStyle.Primary)
+                    .setDisabled(true),
+                new ButtonBuilder()
+                    .setCustomId('next_page')
+                    .setLabel('Next')
+                    .setStyle(ButtonStyle.Primary)
+                    .setDisabled(totalPages <= 1)
+            );
+
         interaction.reply({
-            "embeds": [
+            embeds: [
                 {
-                    "title": `muusik.app Commands`,
-                    "description": `Here are the commands you can use with the muusik.app bot:`,
-                    "color": colors.Muusik,
-                    "fields": fields,
-                },
+                    title: `muusik.app Commands`,
+                    description: `Here are the commands you can use with the muusik.app bot:`,
+                    color: colors.Muusik,
+                    fields: fields,
+                    footer: { text: `Page 1 of ${totalPages}` }
```

Comment:
isn't this supposed to be dynamic? or did we forget to track the page number? lol

Reply:
Page 1 is there because that's the first time you run the command, it's dynamic below when you press buttons

Reply:
ignore me, im dumb :3